Add Mortar Tower, Explosion Effects

Fenway17 opened the pull request:
Added explosion effects particle systems for mortar and gel explosion towers:
- Added explosion particle system (for mortar tower)
  - Added sprites and material for explosion particle system
  - Added gel explosion particle system
    - Added gel explosion elemental variants
    - Changed gel explosion tower to have new hit effects
- ![image](https://user-images.githubusercontent.com/24566006/194285583-2c24b9a0-7c65-481b-9472-9af2a0b54141.png)


Added mortar tower:
- Added base scripts and scriptable objects for mortar tower
  - Added minimum range inside script
- Added mortar bullets using cannon bullets
- Added math scripts to introduce parabola movement to mortar shots
  - *Parabola controller can be used to input more complex parabolas in the future*
- Updated mortar to use correct models and bullets
- Mortar bullets curve properly when shot
- Added sprite images for mortar tower

Miscellaneous:
- Changed some bullet script access modifiers to protected

Eriksen2411 committed:
Tower and Spider Projectile Models

- Added Mortar Tower Model (Models/Towers/Mortar)
- Added Gel Explosive Tower Model (Models/Towers/Gel Explosive)
- Added Spider Projectile Model (Models/Spiders)

Fenway17 committed:
Added base scripts and scriptable objects for mortar tower
Added explosion particle system (for mortar tower)
- Added sprites and material for explosion particle system
- Added gel explosion particle system
  - Added gel explosion elemental variants
  - Changed gel explosion tower to have new hit effects

Fenway17 committed:
Removed explosion particle system gameobject from tutorial level (used for testing)

Fenway17 committed:
Created mortar tower scriptable objects

Fenway17 committed:
Updated mortar tower
- Fixed names of mortar tower
- Updated bullet script to replace public modifiers with protected
- Updated gel bullet script to replace public modifiers with protected
- Added math scripts to introduce parabola movement of mortar bullet
- Added mortar bullet script to use parabola movement

Fenway17 committed:
Updated mortar tower
- Created mortar bullet prefabs using cannon bullets
- Mortar bullets now follow a parabola instead of a straight line

Fenway17 committed:
Updated mortar tower
- Bullet now has no homing; homing effect can be reenabled in the mortar bullet script
- Added placeholder mortar tower based on ballista tower

Eriksen2411 committed:
Update materials

Update (Link) Materials For Mortar Tower

Eriksen2411 committed:
Add Models

Add models for mortar with 3 elements

Eriksen2411 committed:
Add Laser Tower Model

Add Basic Laser Tower Model

Fenway17 committed:
Merge remote-tracking branch 'origin/additional_models' into add-mortar-tower

# Conflicts:
#	Assets/Models/Towers/Towers/Gel Explosive/Gel_Explosive_Tower.fbx.meta
#	Assets/Models/Towers/Towers/Mortar/Mortar_Projectile.fbx.meta
#	Assets/Models/Towers/Towers/Mortar/Mortar_Tower.fbx.meta

Fenway17 committed:
Updated mortar tower
- Mortar now has correct tower models and bullet models
- Updated mortar tower prefab to use new tower models
- Updated mortar scriptable objects to use new mortar shell bullets
- Created new mortar tower shell bullets with new bullet models
  - Required to use rotation workaround as model was not pointing in Z-axis direction
- Updated mortar bullet script to set rotation of mortar shell bullets

Fenway17 committed:
Removed debug log code from mortar bullet scripts

Fenway17 committed:
Balancing mortar tower
- Edited costs and damage of mortar tower and its upgrades
  - Values on "Game Stats (Tower/Enemies)" document

Eriksen2411 reviewed (changes requested):
- Rescale mortar shells to 1/3 to 1/2.
- Need to be rebalance later.
- Errors showing in console while playing:
![image](https://user-images.githubusercontent.com/65208242/195810724-286586dc-03ea-4298-b838-2af2e215e771.png)

Fenway17 committed:
Merge remote-tracking branch 'origin/main-v2' into add-mortar-tower

# Conflicts:
#	Assets/Materials/Towers/Towers/Laser.meta
#	Assets/Materials/Towers/Towers/Laser/Laser_Tower.fbm.meta
#	Assets/Materials/Towers/Towers/Laser/Laser_Tower.fbm/laser_tower.png.meta
#	Assets/Materials/Towers/Towers/Laser/laser_tower.mat
#	Assets/Materials/Towers/Towers/Laser/laser_tower.mat.meta
#	Assets/Models/Spiders/venom_projectile.fbx.meta
#	Assets/Scenes/Dev/SpiderLoadout.unity
#	UserSettings/EditorUserSettings.asset

Fenway17 committed:
Updated mortar tower
- Made tower smaller
- Made mortar bullet smaller
- Reduced max range of mortar
- Added sprite images for mortar tower

Updated laser tower
- Added better sprites images for laser tower to fit the art style

trxe committed:
change tower model read/write settings
